### ISSUE TITLE
[Bug] chunk_kda/fused_recurrent_kda **kwargs silently swallow unknown flags — Kimi K3's use_beta_sigmoid_in_kernel=True is a no-op on 0.5.0, wrong numerics with no warning

### ISSUE THREAD

**Issue by xiaguan** [fla-org/flash-linear-attention#1119]
### Summary

`chunk_kda` / `fused_recurrent_kda` accept a trailing `**kwargs`, so a semantically **required** flag that a given release does not know about is silently discarded instead of raising. Concretely: Kimi K3's published remote modeling code calls both entry points with `use_beta_sigmoid_in_kernel=True`. On fla-core/flash-linear-attention **0.5.0** that parameter does not exist in either signature — the call lands in `**kwargs`, beta is never passed through a sigmoid, and the model computes wrong recurrences **with no error and no warning**. The parameter exists on 0.5.2 (and current `main`), so up-to-date installs are fine, but any environment pinned to 0.5.0 (ours was) runs K3 silently wrong on the HF-transformers + public-fla path.

### How we found it

While building per-boundary numerical oracles for Kimi K3 (bf16, checking a re-implementation against the published modeling code + fla as reference), the KDA boundary refused to converge past cosine ≈ 0.5 against the recurrent path. An exhaustive hypothesis matrix over {scale, state layout, decay axis, beta sigmoid, bias, l2norm} matched the fla 0.5.0 output **bitwise** only with the beta sigmoid *disabled* — i.e. 0.5.0 was not applying it, despite the model code passing `use_beta_sigmoid_in_kernel=True`. Inspecting the 0.5.0 signatures confirmed the kwarg is absorbed by `**kwargs`:

```python
import inspect
from fla.ops.kda import chunk_kda, fused_recurrent_kda   # fla 0.5.0
'use_beta_sigmoid_in_kernel' in inspect.signature(chunk_kda).parameters            # False
'use_beta_sigmoid_in_kernel' in inspect.signature(fused_recurrent_kda).parameters  # False
# both signatures end in **kwargs -> call succeeds, flag ignored
```

The intent is unambiguous (the docstrings on current main require pre-sigmoided beta when the flag is off, and vLLM's KDA implementation applies the sigmoid), so 0.5.0 + K3 remote code = silently wrong numerics.

### Ask

The specific parameter is already fixed on ≥ 0.5.1, but the failure *class* is still open on `main`: both entry points still end in `**kwargs`, so the next semantically-required flag added for a new checkpoint will silently no-op on every older release the same way. Could the public entry points either

1. drop the bare `**kwargs`, so unknown arguments raise `TypeError` (the standard Python contract), or
2. keep it for internal plumbing but `warnings.warn`/raise on any key they do not consume?

A hard error at call time turns "model quietly produces wrong logits" into a one-line version-bump fix for the user. Happy to send a PR for either variant if the maintainers have a preference.

### Environment

- flash-linear-attention 0.5.0 (pip), PyTorch 2.13.0+cu130, Python 3.12, aarch64 + NVIDIA GB300 (sm_103)
- Reference model: Kimi K3 published checkpoint remote code (`modeling_kimi_linear.py`, calls at the `use_beta_sigmoid_in_kernel=True` sites for both `chunk_kda` and `fused_recurrent_kda`)

**Comment by zhiyuan1i** [comment]
It doesn't make sense to fix this issue, as FLA doesn't release patches for previous versions.
